homeassistant/components/automation/trace.py

```diff
@@ -2,148 +2,25 @@
-
-    def set_action_trace(self, trace: dict[str, Deque[TraceElement]]) -> None:
-        """Set action trace."""
-        self._action_trace = trace
-
-    def set_condition_trace(self, trace: dict[str, Deque[TraceElement]]) -> None:
-        """Set condition trace."""
-        self._condition_trace = trace
-
-    def set_error(self, ex: Exception) -> None:
-        """Set error."""
-        self._error = ex
-
-    def set_variables(self, variables: dict[str, Any]) -> None:
-        """Set variables."""
-        self._variables = variables
-
-    def finished(self) -> None:
-        """Set finish time."""
-        self._timestamp_finish = dt_util.utcnow()
-        self._state = "stopped"
-
-    def as_dict(self) -> dict[str, Any]:
-        """Return dictionary version of this AutomationTrace."""
-
-        result = self.as_short_dict()
-
-        action_traces = {}
-        condition_traces = {}
-        if self._action_trace:
-            for key, trace_list in self._action_trace.items():
-                action_traces[key] = [item.as_dict() for item in trace_list]
-
-        if self._condition_trace:
-            for key, trace_list in self._condition_trace.items():
-                condition_traces[key] = [item.as_dict() for item in trace_list]
-
-        result.update(
-            {
-                "action_trace": action_traces,
-                "condition_trace": condition_traces,
-                "config": self._config,
-                "context": self.context,
-                "variables": self._variables,
-            }
-        )
-        if self._error is not None:
-            result["error"] = str(self._error)
-        return result
-
-    def as_short_dict(self) -> dict[str, Any]:
-        """Return a brief dictionary version of this AutomationTrace."""
-
-        last_action = None
-        last_condition = None
-        trigger = None
-
-        if self._action_trace:
-            last_action = list(self._action_trace)[-1]
-        if self._condition_trace:
-            last_condition = list(self._condition_trace)[-1]
-        if self._variables:
-            trigger = self._variables.get("trigger", {}).get("description")
-
-        result = {
-            "last_action": last_action,
-            "last_condition": last_condition,
-            "run_id": self.run_id,
-            "state": self._state,
-            "timestamp": {
-                "start": self._timestamp_start,
-                "finish": self._timestamp_finish,
-            },
-            "trigger": trigger,
-            "domain": self._key[0],
-            "item_id": self._key[1],
-        }
-        if self._error is not None:
-            result["error"] = str(self._error)
-        if last_action is not None:
-            result["last_action"] = last_action
-            result["last_condition"] = last_condition
-
-        return result
-
-
 @contextmanager
 def trace_automation(hass, item_id, config, context):
```

Comment:
if it's automation specific, should we keep calling it `automation_id` ?

Reply:
Fixed in #48288